kernel/mmanutils.c

```diff
@@ -570,9 +634,25 @@ int myst_munmap(void* addr, size_t length)
     /* align length to a page boundary */
     ECHECK(myst_round_up(length, PAGE_SIZE, &length));
 
+    _rlock(&locked);
     ECHECK(myst_mman_munmap(&_mman, addr, length));
+    ECHECK(_remove_file_mappings(addr, length, head_out));
+    _runlock(&locked);
+
+done:
+    _runlock(&locked);
+    return ret;
+}
+
+int myst_munmap(void* addr, size_t length)
+{
+    int ret = 0;
+    fdlist_t* head = NULL;
+
+    ECHECK(__myst_munmap(addr, length, &head));
```

Comment:
any reason the file handles arent returned from the remove_file_mappings instead? At least the handles are related to that, right?
Not a big deal, but could possibly make it more obvious.

Reply:
The file handles are returned by remove_file_mappings, but I added a `__myst_munmap` wrapper so that the munmap and clearing of fdmappings happens atomically.